File: 2017-03-29/W4D4 - Storage/W4D4 - Storage/ViewController.m
//
//  ViewController.m
//  W4D4 - Storage
//
//  Created by Jason Liang on 3/29/17.
//  Copyright © 2017 jasonliang. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UITextField *textField;
@property (weak, nonatomic) IBOutlet UITextField *textField2;

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  
  // 1. retrieving via nsuserdefaults
  NSString *text = [[NSUserDefaults standardUserDefaults] objectForKey:@"name"];
  self.textField.text = text;
  
  
  // 2. retrieving via nsfilemanager
  NSFileManager *fileManager = [NSFileManager defaultManager];
  
  NSArray *docDirectories = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
  NSString *docPath = [docDirectories firstObject];
  NSString *filePath = [NSString stringWithFormat:@"%@/%@", docPath, @"example.txt"];
  
  NSData *data = [fileManager contentsAtPath:filePath];
  NSString *text2 = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
  self.textField2.text = text2;
  
}


- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
  // Dispose of any resources that can be recreated.
}

- (IBAction)save:(id)sender {
  NSString *text = self.textField.text;
  [[NSUserDefaults standardUserDefaults] setObject:text forKey:@"name"];
  NSInteger age = 30;
  [[NSUserDefaults standardUserDefaults] setObject:[NSNumber numberWithInteger:age] forKey:@"age"];
  
  [[NSUserDefaults standardUserDefaults] synchronize];
}

- (IBAction)save2:(id)sender {
  
  NSArray *docDirectories = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
  NSString *docPath = [docDirectories firstObject];
  
  NSString *text = self.textField2.text;
  NSData *data = [text dataUsingEncoding:NSUTF8StringEncoding];
  
  NSString *filePath = [NSString stringWithFormat:@"%@/%@", docPath, @"example.txt"];
  
  
//  1.
  [data writeToFile:filePath atomically:YES];
  
  // 2.
  NSFileManager *fileManager = [NSFileManager defaultManager];
  [fileManager createFileAtPath:filePath contents:data attributes:nil];
  
  

//  NSURL *docURL = [NSURL fileURLWithPath:docPath];
}

@end

File: 2017-12-18/W8D1-Firebase/W8D1-Firebase/ViewController.swift
//
//  ViewController.swift
//  W8D1-Firebase
//
//  Created by Jason Liang on 12/18/17.
//  Copyright © 2017 Jason Liang. All rights reserved.
//

import UIKit
import FirebaseDatabase

class ViewController: UIViewController {
  
  @IBOutlet weak var textField: UITextField!
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
  }
  
  override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning()
    // Dispose of any resources that can be recreated.
  }
  
  @IBAction func addEntryTapped(_ sender: Any) {
    //there's some text for us to enter
    
    guard let text = textField.text else { return }
    if text.count > 0 {
      let ref = Database.database().reference().child("users")
      let key = ref.child("users").childByAutoId().key
      ref.updateChildValues([key:text])
      textField.text = ""
    }
  }
  
}

